public/locations-list-map.js

```diff
@@ -316,10 +367,186 @@ function initMap(stateFilter) {
+ * Strictly responsible for resetting the map to it's initial state on page load WITHOUT user's location (since we have
+ * a link to link to go back to that appearance).
+ */
+function resetMap() {
+  // TODO THIS SHOULD HAVE TWO FUNCTION CALLS INSTEAD OF ONE:
+  //  1.) Show initial markers based on initial filters (e.g. which state[s]?)
+  //  2.) Reposition map (e.g. initial state?)
+  //  However, repositioning is ALSO happening deep inside showMarkers() function which has made itself responsible for
+  //  too many tasks.
+  showMarkers(window.data_by_location, initial_marker_filters, !initial_marker_filters);
+}
+
+
+
+/**
+ * Takes the form input and attempts to pull geocoded information from it. Once done, centers map on closest markers nearby.
+ */
+function loadSearchInMap() {
+  if (!autocomplete) {
+    console.error('autocomplete is not yet defined.');
+    return;
+  }
+
+  let place = autocomplete.getPlace();
+  if (place.geometry) {
+    // Get the location object that we can map.setCenter() on.
+    let location = place.geometry.location;
+    if (location) {
+      centerMapToMarkersNearCoords(location.lat(), location.lng())
+
+    } else {
+      console.warn('Location data not found in place geometry (place.geometry.location).')
+    }
+  } else {
+    console.warn('No geometry found, skipping place');
+  }
+}
+
+
+
+/**
+ * Centers map at automatically detected coordinates using built in navigator.geolocation API.
+ */
+function centerMapToMarkersNearUser() {
+  // First check to see if the user will accept getting their location, if not, silently return
+  if (navigator.geolocation) {
+    navigator.geolocation.getCurrentPosition(function (position) {
+      // Use navigator provided lat/long coords to center map now.
+      centerMapToMarkersNearCoords(position.coords.latitude, position.coords.longitude);
+
+    }, function (err) {
+      // Hide the "User my location" link since we know that will not work.
+      $('#use-location').hide();
+
+    }, {
+      maximumAge: Infinity,
+      timeout: 10000
+    });
+  }
+}
+
+
+
+/**
+ * Centers map around markers nearest to an arbitrary set of latitude/longitude coordinates.
+ */
+function centerMapToMarkersNearCoords(latitude, longitude) {
+  var latlng = new google.maps.LatLng(latitude, longitude);
+
+  //Compute the distances of all markers from the user
+  var markerDistances = new Map(); // an associative array containing the marker referenced by the computed distance
+  var distances = []; // all the distances, so we can sort and then call markerDistances
+  for (const marker of markers) {
+    let distance = google.maps.geometry.spherical.computeDistanceBetween(marker.position, latlng);
+
+    // HACK: In the unlikely event that the exact same distance is computed, add one meter to the distance to give it a unique distance
+    // This could occur if a marker was added twice to the same location.
+    if (markerDistances.has(distance)) { distance = distance + 1; }
```

Comment:
Probably fine to ignore multiple markers at the same exact distance.

Reply:
this is not my code, copied. should add to your list of notes to review once merged.